enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,477 @@
+   The packet will hit the following openflow rules:
+   
+   ```none
+   cookie=0xdeff105, duration=48397.416s, table=0, n_packets=6, n_bytes=928, priority=500,ip,in_port="patch-br-ex_rhe",nw_src=10.44.0.0,nw_dst=10.89.0.3 actions=ct(commit,table=4,zone=64001)
+   cookie=0xdeff105, duration=48397.416s, table=4, n_packets=6, n_bytes=928, ip actions=ct(commit,table=3,zone=64002,nat(src=169.254.169.1))
+   cookie=0xdeff105, duration=10.170s, table=3, n_packets=12647, n_bytes=1517745, actions=move:NXM_OF_ETH_DST[]->NXM_OF_ETH_SRC[],mod_dl_dst:9e:0c:05:73:23:54,LOCAL
+   ```
+
+6. The packet enters the host kernel, then be processed by the hostNetwork pod.
+
+#### Reply
+
+1. the hostNetwork pod sends the reply packet with src_ip(10.89.0.3) and
+   dest_ip(169.254.169.1).
+
+2. In the host routing table there's a static route that forward the traffic to
+   169.254.169.1 via br-ex
+   
+   ```none
+   169.254.169.1 dev br-ex src 10.89.0.3
+   ```
+
+3. In br-ex, the packet will be unDNAT to 10.44.0.0 then forwarded to OVN
+   gateway router
+   ```none
+   cookie=0xdeff105, duration=48397.416s, table=0, n_packets=5, n_bytes=1974, priority=500,ip,in_port=LOCAL,nw_dst=169.254.169.1 actions=ct(table=5,zone=64002,nat)
+   cookie=0xdeff105, duration=48397.416s, table=5, n_packets=5, n_bytes=1974, ip actions=ct(commit,table=2,zone=64001,nat)
+   cookie=0xdeff105, duration=10.170s, table=2, n_packets=16588, n_bytes=1428836, actions=mod_dl_dst:9e:0c:05:73:23:54,output:"patch-br-ex_rhe"
+   ```
+4. The loadbalancer in OVN GR will then handle unSNAT, unDNAT to src_ip
+   (10.43.117.48), dest_ip(169.254.169.2) and send the packet back to br-ex.
+
+5. br-ex will unDNAT and send the packet to localhost.
+   
+   ```none
+   cookie=0xdeff105, duration=13742.995s, table=0, n_packets=2198, n_bytes=306654, priority=500,ip,in_port="patch-br-ex_rhe",nw_src=10.43.0.0/16,nw_dst=169.254.169.2 actions=ct(table=3,zone=64001,nat)
+   cookie=0xdeff105, duration=14.962s, table=3, n_packets=2198, n_bytes=306654, actions=move:NXM_OF_ETH_DST[]->NXM_OF_ETH_SRC[],mod_dl_dst:16:38:a3:9b:fd:6b,LOCAL
+   ```
+   
+### Host to NodePort Services Backed by Host Pods
+
+This traffic case is almost the same as the previous one. The only difference is
+that before the packet is sent to br-ex, it will be DNATed to the service's
+clusterIP by host iptables rule.
+
+```none
+-A OVN-KUBE-NODEPORT -p tcp -m addrtype --dst-type LOCAL -m tcp --dport 30847 -j DNAT --to-destination 10.43.117.48:80
+```
+
+### External to NodePort Services
+
+The general flow is:
+
+1. The packet enters host via eth0.
+2. In the host, the packet is DNATed to the service's clusterIP by the following iptables rule.
+
+```none
+-A OVN-KUBE-NODEPORT -p tcp -m addrtype --dst-type LOCAL -m tcp --dport 30847 -j DNAT --to-destination 10.43.117.48:80
+```
+
+3. The packet is forwarded to br-ex bridge.
+
+The rest of the steps is the same as the normal local gateway mode.
```

Comment:
in all these 3 cases aren't the steps exactly what we do today for lgw mode?
is my understanding correct that the only difference here is we won't enter through `br-ex` since bridge is no longer shared but instead through physical interface?
```
Instead of creating the br-ex bridge with the node's default interface, we
create the br-ex without adding any host interface as its uplink port.
```
is the only new thing right?

Reply:
We also change the default route in the GR. As there will be no default gateway reachable from br-ex.